javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,385 @@
+
+  def pop(): E =
+    removeFirst()
+
+  private def _removeOccurrence(iter: Iterator[E], o: Any): Boolean = {
+    var changed = false
+    val e = o.asInstanceOf[E]
+    while (iter.hasNext() && !changed) {
+      if (iter.next() === e) {
+        iter.remove()
+        changed = true
+      }
+    }
+
+    changed
+  }
+
+  def removeFirstOccurrence(o: Any): Boolean =
+    _removeOccurrence(iterator(), o)
+
+  def removeLastOccurrence(o: Any): Boolean =
+    _removeOccurrence(descendingIterator(), o)
+
+  override def listIterator(index: Int): ListIterator[E] = {
+    checkIndexOnBounds(index)
+    new ListIterator[E] {
+
+      private var last = -1
+      private var i = index
+
+      private var currentNode: Node[E] =
+        if (index == size) null
+        else getNodeAt(index)
+
+      private var lastNode: Node[E] =
+        if (currentNode eq null) getNodeAt(size - 1)
+        else null
+
+      def hasNext(): Boolean =
+        (i < size)
+
+      def next(): E = {
+        if (i > size) throw new NoSuchElementException()
+        last = i
+        i += 1
+
+        if (currentNode eq null)
+          currentNode = lastNode
+
+        lastNode = currentNode
+        currentNode = currentNode.next
+
+        lastNode.value
+      }
+
+      def hasPrevious(): Boolean =
+        (i > 0)
+
+      def previous(): E = {
+        if (i < 0) throw new NoSuchElementException()
+        i -= 1
+        last = i
+
+        if (currentNode eq null)
+          currentNode = lastNode
+        else
+          currentNode = currentNode.prev
+
+        lastNode = currentNode
+
+        lastNode.value
+      }
+
+      def nextIndex(): Int = i
+
+      def previousIndex(): Int = i - 1
+
+      def remove(): Unit = {
+        checkThatHasLast()
+        removeNode(lastNode)
+        if (last < i)
+          i -= 1
+        last = -1
+      }
+
+      def set(e: E): Unit = {
+        checkThatHasLast()
+        lastNode.value = e
+      }
+
+      def add(e: E): Unit = {
+        if (currentNode eq null) {
+          if (lastNode.next eq null) {
+            addLast(e)
+            lastNode = getNodeAt(size - 1)
+          } else if (lastNode.prev eq null) {
+            addFirst(e)
+            lastNode = getNodeAt(0)
+          }
+        } else {
+          addNode(currentNode, e)
+        }
+        last = -1
+        i += 1
+      }
+
+      private def checkThatHasLast(): Unit = {
+        if (last == -1)
+          throw new IllegalStateException()
+      }
+    }
+  }
+
+  def descendingIterator(): Iterator[E] = {
+    new Iterator[E] {
+      private val iter = listIterator(_size)
+
+      def hasNext(): Boolean =
+        iter.hasPrevious
+
+      def next(): E =
+        iter.previous
+
+      def remove(): Unit =
+        iter.remove
+
+    }
+  }
+
+  override def clone(): AnyRef =
+    new LinkedList[E](this)
+
+}
+
+object LinkedList {
+
+  protected[LinkedList] final class Node[T](
+      var value: T = null.asInstanceOf[T],
```

Comment:
This default value is not necessary, because you never create a `Node` without explicitly specifying the first parameter.